File: spec/php/NavParent2Test.php
<?php
namespace Kaitai\Struct\Tests;

class NavParent2Test extends TestCase {
    public function testNavParent2() {
        $r = NavParent2::fromFile(self::SRC_DIR_PATH . "/nav_parent2.bin");

        $this->assertEquals(8, $r->ofsTags);
        $this->assertEquals(2, $r->numTags);

        $this->assertEquals('RAHC', $r->tags[0]->name);
        $this->assertEquals(0x20, $r->tags[0]->ofs);
        $this->assertEquals(3, $r->tags[0]->numItems);
        $this->assertEquals('foo', $r->tags[0]->tagContent->content);

        $this->assertEquals('RAHC', $r->tags[1]->name);
        $this->assertEquals(0x23, $r->tags[1]->ofs);
        $this->assertEquals(6, $r->tags[1]->numItems);
        $this->assertEquals('barbaz', $r->tags[1]->tagContent->content);
    }
}

File: spec/zig/tests/nav_parent2_test.zig
const _imp_std = @import("std");
const _imp_kaitai_struct = @import("kaitai_struct");
const _imp_nav_parent2 = @import("../formats/nav_parent2.zig");

test "NavParent2" {
    const file = try _imp_std.fs.cwd().openFile("../../src/nav_parent2.bin", .{});
    defer file.close();
    var buffer: [8]u8 = undefined;
    var reader = file.reader(&buffer);
    const allocator = _imp_std.testing.allocator;
    var arena = _imp_std.heap.ArenaAllocator.init(allocator);
    defer arena.deinit();
    var _io = _imp_kaitai_struct.KaitaiStream.fromFileReader(&reader);
    const r = try _imp_nav_parent2.NavParent2.create(&arena, &_io, null, null);
    try _imp_std.testing.expectEqual(8, r.ofs_tags);
    try _imp_std.testing.expectEqual(2, r.num_tags);
    try _imp_std.testing.expectEqualStrings("RAHC", r.tags.items[0].name);
    try _imp_std.testing.expectEqual(32, r.tags.items[0].ofs);
    try _imp_std.testing.expectEqual(3, r.tags.items[0].num_items);
    try _imp_std.testing.expectEqualStrings("foo", (try r.tags.items[0].tagContent()).?.content);
    try _imp_std.testing.expectEqualStrings("RAHC", r.tags.items[1].name);
    try _imp_std.testing.expectEqual(35, r.tags.items[1].ofs);
    try _imp_std.testing.expectEqual(6, r.tags.items[1].num_items);
    try _imp_std.testing.expectEqualStrings("barbaz", (try r.tags.items[1].tagContent()).?.content);
}
